src/components/browser/requestOperations/Swap.tsx

```diff
@@ -0,0 +1,214 @@
+          extendedAccount,
+        );
+        if (tokenBalances && tokenBalances.length > 0) {
+          const tokenBalance = tokenBalances.find(
+            (token) => token.symbol === startToken,
+          );
+          if (tokenBalance) {
+            balance = parseFloat(tokenBalance.balance);
+          }
+        }
+      }
+
+      return balance;
+    } catch (error) {
+      showModal('request.error.swap.fetching_balance', MessageModalType.ERROR, {
+        error: error.message,
+      });
+      return 0;
+    }
+  };
+
+  useEffect(() => {
+    const initializeData = async () => {
+      try {
+        const balance = await getBalance();
+        setBalance(balance);
+        const newBalanceAfterSwap = balance - amount;
+        setBalanceAfterSwap(newBalanceAfterSwap);
+      } catch (error) {
+        console.error('Error initializing data:', error);
+        showModal(
+          'request.error.swap.fetching_balance',
+          MessageModalType.ERROR,
+          {
+            error: error.message,
+          },
+        );
+      }
+    };
+
+    initializeData();
+  }, [startToken, username, accounts, amount]);
+
+  const handleSwap = async (options: TransactionOptions) => {
+    try {
+      if (balanceAfterSwap < 0) {
+        throw new Error(translate('request.error.swap.insufficient_balance'));
+      }
+
+      const swapConfig = await SwapTokenUtils.getConfig();
+
+      const account = accounts.find((e) => e.name === request.username);
+      if (!account) {
+        throw new Error('Account not found');
+      }
+      if (!swapConfig) {
+        throw new Error('Failed to load swap configuration');
+      }
+
+      const estimateId = await SwapTokenUtils.saveEstimate(
+        steps,
+        slippage,
+        startToken,
+        endToken,
+        amount,
+        username,
+      );
+      if (!estimateId) {
+        throw new Error('Failed to save swap estimate');
+      }
+
+      const result = await SwapTokenUtils.processSwap(
+        estimateId.toString(),
+        startToken,
+        amount,
+        (account as unknown) as ActiveAccount,
+        swapConfig.account,
+        options,
+      );
+
+      if (!result) {
+        throw new Error('Swap transfer failed');
+      }
+
+      return {...result, swap_id: estimateId};
+    } catch (error) {
+      if (error.message) {
+        throw new Error(`Swap failed: ${error.message}`);
+      } else {
+        throw new Error('Swap failed: Unknown error occurred');
+      }
+    }
+  };
+
+  return (
+    <RequestOperation
+      sendResponse={sendResponse}
+      sendError={sendError}
+      successMessage={translate('request.success.swap', {
+        amount,
+        from: startToken,
+        to: endToken,
+      })}
+      errorMessage={translate('request.error.swap.failed')}
+      beautifyError
+      method={KeyTypes.active}
+      request={request}
+      closeGracefully={closeGracefully}
+      performOperation={handleSwap}>
+      <RequestItem
```

Comment:
This can be an anonymous request so this should be using the `usePotentiallyAnonymousRequest` hook. See usage on other Requests.